clients/drcachesim/reader/v2p_reader.h

```diff
@@ -0,0 +1,61 @@
+/* **********************************************************
+ * Copyright (c) 2024 Google, LLC  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, LLC nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, LLC OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* v2p_reader: reads and parses a virtual-to-physical address mapping in textproto format.
+ * Creates virtual-to-physical address map in memory.
+ */
+
+#ifndef _V2P_READER_H_
+#define _V2P_READER_H_ 1
+
+#include <stdint.h>
+
+#include <cstdint>
+#include <unordered_map>
+#include <string>
+
+namespace dynamorio {
+namespace drmemtrace {
+
+class v2p_reader_t {
+public:
+    v2p_reader_t();
+
+    bool
+    gen_v2p_map(std::string path_to_file,
+                std::unordered_map<uint64_t, uint64_t> &v2p_map);
```

Comment:
Is the plan to add a use of this class in the TLB simulator?  Add that to the PR description?  Have you prototyped that usage yet?  I assume it needs the page size as the prior comment says??

Reply:
Yes, the plan is to use this reader in the TLB simulator.
Added to the PR description.
Usage prototyped here: https://github.com/DynamoRIO/dynamorio/pull/6961 (almost ready for review).
Yes, page size added.